pkg/processor/pending_request.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2025 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	allocationpb "agones.dev/agones/pkg/allocation/go"
+)
+
+// pendingRequest represents a request waiting for processing.
+type pendingRequest struct {
```

Comment:
This could probably be pulled into processor.go ? not sure it needs it's own file.